### PR TITLE
Update recipe script

### PR DESCRIPTION
**Pull request by Pahandrovich**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -21,14 +21,14 @@ requirements:
     - python
     - setuptools
     - numpy
-    - daal-devel >=2020.3
+    - daal-devel >=2021.1
```
Comment (PetrovKP) [inline recipe/meta.yaml:24]:
```suggestion
    - dal-devel {{ version }}
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -21,14 +21,14 @@ requirements:
...
     - cython
     - jinja2
     - mpich      # [not win]
     - clang-tools
   run:
     - python
-    - daal >=2020.3
+    - daal >=2021.1
```
Comment (PetrovKP) [inline recipe/meta.yaml:31]:
```suggestion
    - dal {{ version }}
```

**Comment by PetrovKP** [comment]
We don't have the oneDAL 2021.1 package in the conda-forge repository yet. So CI will be red

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-{% set version = "2020.3" %}
+{% set version = "2021.1" %}
```
Comment (napetrov) [inline recipe/meta.yaml:1]:
```suggestion
{% set version = "2020.3.1" %}
```

File: recipe/meta.yaml [hunk 4]
```diff
@@ -7,7 +7,7 @@ package:
 
 source:
   url: https://github.com/IntelPython/daal4py/archive/{{ version }}.tar.gz
-  sha256: 5926d4e187ae1cc8fbd20b2d6026be8744fb8ea0603cee857e78295c206b46b6
+  sha256: f0c9e82490cc865e039137c27ed37bc46f17db64594ca80c694ab466e2470903
```
Comment (napetrov) [inline recipe/meta.yaml:10]:
```suggestion
  sha256: 2b067b1d2fc6addcadbe0f5271b151c3d050b44a16e723656acf2521ec2ff729
```

**Comment by napetrov** [comment]
@conda-forge-admin, please rerender

**Comment by PetrovKP** [comment]
@conda-forge-admin, please rerender

**Comment by PetrovKP** [comment]
@conda-forge-admin, please restart ci

**Review by napetrov — APPROVED** [review]

**Pull request merged**